disk/test/com/treode/disk/edit/DiskIOSpec.scala

```diff
@@ -18,62 +18,150 @@ package com.treode.disk.edit
 
 import com.treode.async.io.stubs.StubFile
 import com.treode.async.stubs.StubScheduler
-import com.treode.async.stubs.implicits._
 import com.treode.disk.DiskTestConfig
 import org.scalatest.FlatSpec
+import com.treode.pickle.Picklers._
+import com.treode.disk.PageDescriptor
+import com.treode.disk.Position
+import com.treode.async.{Async, Callback, Fiber, Scheduler }, Async.async
+import com.treode.disk.PickledPage
+import com.treode.async.stubs.implicits._
+
 
 class DiskIOSpec extends FlatSpec {
 
   implicit val config = DiskTestConfig()
-
-  "The PageReader" should "be able to read the string the PageWriter wrote" in {
+   "The PageReader" should "be able to read the tuple the PageWriter wrote" in {
     implicit val scheduler = StubScheduler.random()
-    val a = "this is a string"
+    val stringPickler = 
+        wrap(string,string)
+        .build(x => (x._1, x._2) )
+        .inspect( x=> (x._1, x._2) )
+
+    val a = "this is "
+    val b = "a string!"
     val readPos = 0
     val f = StubFile (1 << 20, 0)
-
-    val dw = new PageWriter (f)
+    val dsp = new PageDispatcher
+    val dw = new PageWriter (dsp, f)
     val dr = new PageReader (f)
-    val (pos, len) = dw.write (a) .expectPass()
-    val readString = dr.readString (readPos, len) .expectPass()
-    assert (a.equals (readString))
+    
+    val pDesc = PageDescriptor(0x25, stringPickler)
+    val posA = dsp.write(pDesc, 0, 0, (a,b) ) .expectPass()
+    val readString = dr.read(pDesc, posA).expectPass()
+    assert ((a,b).equals (readString))
   }
 
-  it should "write and read multiple times to disk" in {
+  "The DiskIO Reader/Writer " should " be able to write and read multiple times to disk" in {
     implicit val scheduler = StubScheduler.random()
+
+    val stringPickler = 
+        wrap(string)
+        .build(x => x )
+        .inspect( x=> x )
+
     val a = "abcdef"
     val b = "123456789"
     val startPos = 0
     val f = StubFile (1 << 20, 0)
-
-    val dw = new PageWriter (f)
+    val dsp = new PageDispatcher
+    val dw = new PageWriter (dsp,f)
     val dr = new PageReader (f)
-    val (posA, lenA) = dw.write (a) .expectPass()
-    val (posB, lenB) = dw.write (b) .expectPass()
-    val readA = dr.readString (startPos, lenA) .expectPass()
-    val readB = dr.readString (posA,     lenB) .expectPass()
+    val pDesc = PageDescriptor(0x25, stringPickler)
+    val posA = dsp.write(pDesc, 0, 0, a ) .expectPass()
```

Comment:
Again, save yourself from all the temporary identifiers by saying

    dsp.write(pDesc, 0, 0, a ) .expectPass (startPos + lenA)